src/oc/math.cpp

```diff
@@ -61,6 +65,120 @@ double hoc_Log10(double x) {
     return errcheck(log10(x), "log10");
 }
 
+#if NRN_ARCH_INDEP_EXP_POW
+
+static double accuracy32(double val) {
+    int ex;
+    double mant = frexp(val, &ex);
+    // round to about 32 bits after .
+    double prec = 4294967296.0;
+    double result = mant * prec;
+    result = round(result);
+    result /= prec;
+    result = ldexp(result, ex);
+    return result;
+}
+
+static double pow_arch_indep(double x, double y) {
+    mpfr_prec_t prec = 53;
+    mpfr_rnd_t rnd = MPFR_RNDN;
+
+    mpfr_t x_, y_;
+    mpfr_init2(x_, prec);
+    mpfr_init2(y_, prec);
+
+    mpfr_set_d(x_, x, rnd);
+    mpfr_set_d(y_, y, rnd);
+
+    mpfr_pow(x_, x_, y_, rnd);
+
+    double r = mpfr_get_d(x_, rnd);
+
+    mpfr_clear(x_);
+    mpfr_clear(y_);
+
+    return r;
+}
+
+static double exp_arch_indep(double x) {
+    mpfr_prec_t prec = 53;
+    mpfr_rnd_t rnd = MPFR_RNDN;
+
+    mpfr_t x_;
+    mpfr_init2(x_, prec);
+    mpfr_set_d(x_, x, rnd);
+    mpfr_exp(x_, x_, rnd);
+    double r = mpfr_get_d(x_, rnd);
+    mpfr_clear(x_);
+    return r;
+}
+
+static double pow_precision32(double x, double y) {
+    return accuracy32(pow(x, y));
+}
+
+static double exp_precision32(double x) {
+    return accuracy32(exp(x));
+}
+
+static double (*pow_ptr)(double x, double y) = pow;
+static double (*pow_ieee_ptr)(double x,
+                              double y) = pow;  // avoid error! Maybe because of c++ overloading.
+static double (*exp_ptr)(double x) = exp;
+
+int nrn_use_exp_pow_precision(int style) {
+    int s = 0;
+    if (style == 0) {  // default IEEE
+        pow_ptr = pow;
+        exp_ptr = exp;
+    } else if (style == 1) {  // 53 bit mpfr
+        pow_ptr = pow_arch_indep;
+        exp_ptr = exp_arch_indep;
+    } else if (style == 2) {  // 32 bit truncation
+        pow_ptr = pow_precision32;
+        exp_ptr = exp_precision32;
+    }
+    if (pow_ptr == pow_ieee_ptr) {
+        s = 0;
+    }
+    if (pow_ptr == pow_arch_indep) {
+        s = 1;
+    }
+    if (pow_ptr == pow_precision32) {
+        s = 2;
+    }
+    nrn_assert(s >= 0 && s <= 2);
+    return s;
```

Comment:
Can `style` be a value other than `0, 1, 2`? If yes, this is fine. If not, it can be simplified down to:
```suggestion
    return style;
```

Reply:
I agree. Only ```return style;``` is needed